packages/ember-glimmer/lib/utils/process-args.js

```diff
@@ -13,23 +14,26 @@ export function processComponentArgs(namedArgs) {
   let props = Object.create(null);
   let args = Object.create(null);
 
-  props[ARGS] = args;
-
+  let seen = {};
   for (let i = 0; i < keys.length; i++) {
     let name = keys[i];
+
+    assert(`Component attribute \`${name}\` is declared multiple times.`, !seen[name]);
```

Comment:
We also need to include information about the name of the component and source template if possible. 

Getting the source template will likely prove to be difficult in JS-land, but I believe an AST transform could catch the most common cases...

Reply:
added component name, but unsure how to get source :P